### PR TITLE
feat(Core): Allow localized motd

### LINKED ISSUE #17911: Feature: Allow localized MOTD
### Describe your feature request or suggestion in detail

Allow to localize the motd table

### Describe a possible solution to your feature or suggestion in detail

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Exitare**
## Changes Proposed:

This PR proposes changes to:
-  [x] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes #17911 

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue
- [x] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. Apply the sql that adjusts the motd table
2. Prefill table with new localized motds and check motd by either logging in / reloading the table / running the ingame command .server motd

## Known Issues and TODO List:

- [x] Default server command is currently only updating enUS locale
- [x] Questions remain about how to handle missing locale data. Fallback to just enUS. What if there is no enUS locale available?
- [x] Additionally, how should OnMotdChange be handled? Technically the updated motd can be propagated, but without any further knowledge that could be any locale. I´d argue that the OnMotdChange function should be adjusted to also provide the locale.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Nyeriah** [comment]
Quick copilot review: 

The provided SQL script modifies the `motd` table by adding several language-specific columns and migrating the existing `text` column data to the `enUS` column. Let's analyze this for database normalization compliance:

1. **Adding Columns**: The script adds columns for various languages (`enUS`, `koKR`, `frFR`, `deDE`, `zhCN`, `zhTW`, `esES`, `esMX`, `ruRU`).

2. **Data Migration**: It moves data from the `text` column to the `enUS` column.

3. **Dropping Column**: The `text` column is then dropped.

To ensure compliance with database normalization rules, especially the Third Normal Form (3NF), the following considerations should be made:

- **1NF (First Normal Form)**: Each table should have atomic columns. Adding multiple language-specific columns may violate this as it introduces redundancy.
- **2NF (Second Normal Form)**: All non-key attributes must be fully functional dependent on the primary key. The current structure may lead to partial dependencies if the table's primary key is not composite.
- **3NF (Third Normal Form)**: Non-key attributes should not have transitive dependencies. Adding multiple language columns can lead to transitive dependencies.

### Recommendations
- **Normalization**: Consider creating a separate table for localized messages to avoid redundancy and ensure proper normalization. For example:

  ```sql
  CREATE TABLE `motd` (
    `id` INT PRIMARY KEY,
    `default_text` LONGTEXT
  );

  CREATE TABLE `motd_localized` (
    `motd_id` INT,
    `locale` VARCHAR(5),
    `localized_text` LONGTEXT,
    PRIMARY KEY (`motd_id`, `locale`),
    FOREIGN KEY (`motd_id`) REFERENCES `motd`(`id`)
  );
  ```

This structure will ensure that the database is normalized and follows best practices for managing localized content.

**Comment by Exitare** [comment]
Any suggestions regarding the commands affected by this PR?

**Comment by Nyeriah** [comment]
Yes, they will crash if used from console. Handler in the command context is not guaranteed to be a player or have a valid session, so this needs to be checked

**Comment by Exitare** [comment]
Thank you for this insight. I was referring to the command .server set motd [optional realm id] [message]. 
and the sScriptMgr->OnMotdChange(motdMessage);

My thoughts would be:
1. The command should be adjusted to also include the locale as an optional
2. The OnMotdChange() should also include the locale that got changed. As a results the function call would use 2 params instead of one.

**Comment by Nyeriah** [comment]
Sure, it can be added as an option, and it should default to english

File: src/server/game/Scripting/ScriptDefines/WorldScript.h
Comment (Kitzunu) [inline src/server/game/Scripting/ScriptDefines/WorldScript.h:23]: include order

File: src/server/scripts/Commands/cs_reload.cpp
Comment (Kitzunu) [inline src/server/scripts/Commands/cs_reload.cpp:417]: Why not use DEFAULT_LOCALE constant for the "default"?

File: src/server/scripts/Commands/cs_reload.cpp
Comment (Kitzunu) [inline src/server/scripts/Commands/cs_reload.cpp:417]: I dont think we need to use auto here. We know it will always be LocaleConstant anyways

**Comment by Kitzunu** [comment]
If you want some pointers to how I implemented a locale table for quest_greetings check
https://github.com/azerothcore/azerothcore-wotlk/pull/10526 (original implementation)
https://github.com/azerothcore/azerothcore-wotlk/pull/19615 (to make it actually function ingame lmao)

**Comment by Exitare** [comment]
Do you want me to mirror the referenced implementation? The motd loader right now does not work that way and it would be a medium addition at least. Just want to make sure I understand your intentions correctly. As of now, the system works fine.

**Comment by Exitare** [comment]
@Kitzunu I updated to code to reflect your code suggestions

File: src/server/game/Motd/MotdMgr.h
Comment (Kitzunu) [inline src/server/game/Motd/MotdMgr.h:55]: std::string const&

File: src/server/scripts/Commands/cs_server.cpp
Comment (Kitzunu) [inline src/server/scripts/Commands/cs_server.cpp:546]: I feel like the default should always be DEFAULT_LOCALE

File: src/server/scripts/Commands/cs_server.cpp
Comment (Kitzunu) [inline src/server/scripts/Commands/cs_server.cpp:571]: brackets on newline

File: src/server/game/Motd/MotdMgr.cpp
Comment (Kitzunu) [inline src/server/game/Motd/MotdMgr.cpp:124]: so, since we removed the connection between the tables in the database. We need a check in load that there is a "base" MOTD in the first table that corresponds with the entry in the locale table.

Also bracket on newline

File: src/server/scripts/Commands/cs_server.cpp
Comment (Kitzunu) [inline src/server/scripts/Commands/cs_server.cpp:292]: same here, always LocaleConstant

File: src/server/scripts/Commands/cs_server.cpp
Comment (Kitzunu) [reply src/server/scripts/Commands/cs_server.cpp:546]: I feel this should go throughout the entire PR

**Comment by Kitzunu** [comment]
The code is so well documented. I am impressed

File: src/server/game/Motd/MotdMgr.cpp
Comment (Exitare) [reply src/server/game/Motd/MotdMgr.cpp:124]: Sorry, I dont follow. The base motd is always the en_US one, at least that is what was agreed on before. Or are you referring to the realmIds -1 etc? Would you mind providing more details ? Thank you

**Comment by Nyeriah** [comment]
Please keep the MOTD encoding, it's an AC thing to difficult it being removed
Also please check for redundant comments, self-explanatory code doesn't need to be commented (e.g avoid rewriting in English what the code already clearly does)

**Review by Nyeriah — APPROVED** [review]

File: src/server/apps/worldserver/RemoteAccess/RASession.cpp
Comment (Kitzunu) [inline src/server/apps/worldserver/RemoteAccess/RASession.cpp:79]: I feel like this should be DEFAULT_LOCALE too

File: src/server/scripts/Commands/cs_server.cpp
Comment (Kitzunu) [inline src/server/scripts/Commands/cs_server.cpp:292]: this too

File: src/server/scripts/Commands/cs_server.cpp
Comment (Kitzunu) [inline src/server/scripts/Commands/cs_server.cpp:551]: This should be "enUS" without an underscore, right?

File: src/server/scripts/Commands/cs_server.cpp
Comment (Kitzunu) [inline src/server/scripts/Commands/cs_server.cpp:572]: newline for bracket

**Review by Kitzunu — COMMENTED** [review]
I think this is good to go if you address these small things

File: data/sql/updates/pending_db_world/rev_1731983948814732800.sql
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1731983948814732800.sql:4]:
```suggestion
UPDATE `acore_string` SET `content_default` = 'Message of the day in realm {} and locale {} changed to:\r {}', `locale_deDE` = 'Nachricht des Tages für Realm {} und Sprache {} wurde geändert zu:\r {}', `locale_zhCN` = '每日消息更改为 in realm {} and locale {}:\r {}' WHERE (`entry` = 1101);
```

**Comment by Exitare** [comment]
Code adjusted. Should be good to go.

**Review by Kitzunu — APPROVED** [review]

**Comment by Kitzunu** [comment]
Did you test this after the changes were made? If so we can merge

**Comment by Exitare** [comment]
Yes, its tested and working. I had to fix the server command, but now everything is good to go.

**Comment by Kitzunu** [comment]
Awesome. We can merge after DB Squash.

Great job :D

**Comment by Kitzunu** [comment]
CI failing

**Comment by Exitare** [comment]
Already on it ;)

**Comment by Kitzunu** [comment]
Are you in the discord?

**Comment by Exitare** [comment]
Yes. I am the japanese name avatar you responded earlier with the spell dbc in the dev-core channel ;)

**Comment by Kitzunu** [comment]
I see I see

**Pull request merged**
